t/http-cookiejar.t

```diff
@@ -92,6 +93,38 @@ $jar.add-from-response($resp, Cro::Uri.parse: 'http://example.com');
 my $req = Cro::HTTP::Request.new(method => 'GET', target => '/');
 $jar.add-to-request($req, Cro::Uri.parse: 'http://example.com/');
 like $req.Str, /Cookie/,  'Header was added';
-like $req.Str, /'Foo=Bar; Bar=Baz' || 'Bar=Baz; Foo=Bar'/,  'Setted string is correct';
+like $req.Str, /'Foo=Bar; Bar=Baz' || 'Bar=Baz; Foo=Bar'/,  'Set string is correct';
+
+$jar.clear;
+$jar.add-cookie(
+  Cro::HTTP::Cookie.new(
+    name    => 'Test',
+    value   => 'Cookie',
+    max-age => Duration.new(3600e0),
+    domain  => 'example.com',
+    path    => '/'
+  )
+);
+ok $jar.contents.elems == 1,                        'A single cookie was added, successfully';
+ok $jar.contents[0].cookie.name   eq 'Test',        'Added cookie has correct name';
+ok $jar.contents[0].cookie.value  eq 'Cookie',      'Added cookie has correct value';
+ok $jar.contents[0].cookie.domain eq 'example.com', 'Added cookie has proper domain';
+ok $jar.contents[0].cookie.path   eq '/',           'Added cookie has proper path';
+ok DateTime.now.later(seconds => 3600) - $jar.contents[0].expiry-time <= 1,
+  'Cookie has proper expiration time';
+# Add cookie with no expiry
+$jar.add-cookie(
+  Cro::HTTP::Cookie.new(
+    name   => 'Test1',
+    value  => 'Cookie',
+    domain => 'example.com',
+    path   => '/'
+  )
+);
+# Retrieve cookie from jar and check that the expiry is set 10 years later
+ok $jar.contents.elems == 2, 'A second cookie was added successfully';
+ok $jar.contents[1].persistent.not, 'New cookie is not persistent';
+ok DateTime.now.later(years => 10) - $jar.contents[1].expiry-time <= 1,
+  'New cookie has expected expiration time with no max-age set';
```

Comment:
One more test making sure that if we use `add-to-request` then the request does indeed get the cookies added with `add-cookie` would be a good thing.